core/controllers/skill_editor.py

```diff
@@ -146,6 +146,70 @@ class EditableSkillDataHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'skill_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'version': {
+                'schema': {
+                    'type': 'int',
+                }
+            },
+            'commit_message': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'change_dicts': {
+                'schema': {
+                    'type': 'list',
+                    'items': {
+                        'type': 'dict',
+                        'properties': [
+                            {
+                                'name': 'cmd',
+                                'schema': {
+                                    'type': 'basestring'
+                                }
+                            },
+                            {
+                                'name': 'property_name',
+                                'schema': {
+                                    'type': 'basestring'
+                                }
+                            },
+                            {
+                                'name': 'old_value',
+                                'schema': {
+                                    'type': 'basestring'
+                                }
+                            },
+                            {
+                                'name': 'new_value',
+                                'schema': {
+                                    'type': 'basestring'
+                                }
+                            }
```

Comment:
Will the command always have this format?

Reply:
Yeah, you will need to introduce `from_dict` to the `BaseChange` that will load any change from an dictionary and then validates it and the you can use `object_dict` and `object_class` here.

Reply:
Yes, that's what I thought. But I don't know why I am getting this weird error:

> Schema validation for 'change_dicts' failed: from_dict() missing 1 required positional argument: 'base_change_dict'

The signature of from_dict is `def from_dict(self, base_change_dict):`

The schema looks like 
`'change_dicts': {
                'schema': {
                    'type': 'list',
                    'items': {
                        'type': 'object_dict',
                        'object_class': (
                            skill_domain.SkillChange)
                    }
                }
            }`

I looked at other handlers too but they seem similar to what I have written.

Reply:
@vojtechjelinek Can you help with this issue?

Reply:
Where do you see the mentioned error ("Schema validation for 'change_dicts' failed")? I only see "Schema validation for 'change_dicts' failed: validate"

Reply:
Yes, I found the problem and resolved it. Sorry.